dpc/src/circuits/inner_circuit.rs

```diff
@@ -772,6 +825,43 @@ impl<N: Network> ConstraintSynthesizer<N::InnerScalarField> for InnerCircuit<N>
 
                 output_values.push(given_value);
             };
+
+            // ********************************************************************
+            // Check that the ouput value commitment is derived correctly.
+            // ********************************************************************
+            {
+                let vc_cs = &mut cs.ns(|| "Check that value commitment is derived correctly");
+
+                let value_commitment_randomness_gadget = <N::ValueCommitmentGadget as CommitmentGadget<
+                    N::ValueCommitmentScheme,
+                    N::InnerScalarField,
+                >>::RandomnessGadget::alloc(
+                    &mut vc_cs.ns(|| format!("Output value commitment randomness {}", j)),
+                    || Ok(value_commitment_randomness),
+                )?;
+
+                let given_value_commitment_gadget = <N::ValueCommitmentGadget as CommitmentGadget<
+                    N::ValueCommitmentScheme,
+                    N::InnerScalarField,
+                >>::OutputGadget::alloc_input(
+                    &mut vc_cs.ns(|| format!("Output value commitment {}", j)),
+                    || Ok(value_commitment),
+                )?;
+
+                let value_bytes =
+                    UInt8::alloc_vec(vc_cs.ns(|| format!("Output value bytes {}", j)), &record.value().to_bytes_le()?)?;
```

Comment:
This witness is unchecked and technically allows sidestepping the actual record value.

Reply:
I've switched it to the derived bytes